### ISSUE TITLE
[Float8] Float8 rowwise with vanilla TP encounters NaN around 80 iters in...

### ISSUE THREAD

**Issue by lessw2020** [pytorch/torchtitan#865]
### Bug description

Training at 256 scale, float8 rowwise ended up crashing due to NaN's for the loss.  As a control, the same run was done with BF16 and did not encounter any issue. 
Note that 2 runs were done with Float8, both exhibited exact same result at same step. 

Here's a visual:

<img width="501" alt="Image" src="https://github.com/user-attachments/assets/6babda72-e4c0-4885-822a-05f795834620" />

### Versions

Latest TorchAO nightly + TP PR + PyTorch nightly (Feb 10)
TP PR = https://github.com/pytorch/torchtitan/pull/808/files + 
https://github.com/pytorch/ao/pull/1718
This was at 256 scale, with full AC
TP=2 + FP8 rowwise

**Comment by danielvegamyhre** [comment]
@lessw2020 quick question, by "TP PR" are you referring to the torchtitan PR here https://github.com/pytorch/torchtitan/pull/808 or the torchao PR here https://github.com/pytorch/ao/pull/1718? Just want to confirm, were you using both?

**Comment by danielvegamyhre** [comment]
@lessw2020 also was this full AC?

**Comment by lessw2020** [comment]
Hi @danielvegamyhre  - yes to both PR's actually.  I've updated the original issue to mark this.
also yes with full AC.  Let me update the issue as well!

**Comment by danielvegamyhre** [comment]
Hmm, didn't repro in a 100 step run w/ llama3 70b on 128 h100s. Will try a longer one.

**Comment by danielvegamyhre** [comment]
Started by trying to repro locally. 

Confirmed does not repro at 4 GPU scale over 200 steps.

**Comment by danielvegamyhre** [comment]
@lessw2020 IIRC recently you found NaNs occurring for other configurations (not using float8) which led to the conclusion it was an infra issue with the cluster being used, so I think we can close this issue now.

**Comment by lessw2020** [comment]
have not encountered any more NaNs since we isolated  and rebooted a culprit GPU that was issuing XID errors.  
Thus closing as this appears to have been hardware specific.
